test/test-template-engine.js

```diff
@@ -459,10 +459,40 @@ describe('html rendering', () => {
       `;
     };
     const container = document.createElement('div');
-    render(container, getTemplate({ items: [1, 2, 3] }));
-    assert(container.querySelector('#target').childElementCount === 3);
-    render(container, getTemplate({ items: [1, 2, 3, 4, 5] }));
-    assert(container.querySelector('#target').childElementCount === 5);
+
+    // Render 0 > 1 > 2
+    render(container, getTemplate({ items: [] }));
+    assert(container.querySelector('#target').childElementCount === 0);
+    render(container, getTemplate({ items: [1] }));
+    assert(container.querySelector('#target').childElementCount === 1);
+    render(container, getTemplate({ items: [1, 2] }));
+    assert(container.querySelector('#target').childElementCount === 2);
+
+    // Render 2 > 1 > 0
+    render(container, getTemplate({ items: [1, 2] }));
+    assert(container.querySelector('#target').childElementCount === 2);
+    render(container, getTemplate({ items: [1] }));
+    assert(container.querySelector('#target').childElementCount === 1);
+    render(container, getTemplate({ items: [] }));
+    assert(container.querySelector('#target').childElementCount === 0);
+
+    // Render 0 > 1 > 0 > 1
+    render(container, getTemplate({ items: [] }));
+    assert(container.querySelector('#target').childElementCount === 0);
+    render(container, getTemplate({ items: [1] }));
+    assert(container.querySelector('#target').childElementCount === 1);
+    render(container, getTemplate({ items: [] }));
```

Comment:
Specifically flipping back-and-fourth to the zero length case causes one of the issues.